### PR TITLE
Feature/transfer

### PR DESCRIPTION
**Pull request by matprst**
I had something quite complicated but after the reorganization, I realized that integrating it to ```train.py``` was simpler than I thought.
I did not test it with a pretrained ```pure-gnn``` nor with a ```roberta-gnn```.

### CODE REVIEW

File: models/model.py [hunk 1]
```diff
@@ -30,7 +30,9 @@ def __init__(self, model_hparams, optimizer_hparams):
 
         roberta_output_dim = 768
         model_name = model_hparams['model']
-        if model_name == 'roberta':
+        if 'checkpoint' in model_hparams and model_hparams['checkpoint'] is not None:
+            self.model = load_pretrained_encoder(model_hparams['checkpoint'])
```
Comment (rahelbeloch) [inline models/model.py:34]: So we have the same functionality already in the `train.py` with the resume parameter (have you seen that)? I don't think we should have it two times.

Comment (rahelbeloch) [reply models/model.py:34]: So maybe check how you can unify the model loading with resume and the checkpoint parameter and make 1 functionality out of it.

File: train.py [hunk 2]
```diff
@@ -45,10 +45,11 @@ def train(model_name, seed, epochs, patience, b_size, l_rate, w_decay, warmup, m
...
         'cf_hid_dim': cf_hidden_dim,
+        'checkpoint': pretrained,
         **additional_params
     }
 
-    trainer = initialize_trainer(epochs, patience, model_name, l_rate, w_decay, warmup, seed, data_name)
+    trainer = initialize_trainer(epochs, patience, model_name, l_rate, w_decay, warmup, seed, data_name, pretrained)
 
     # optionally resume from a checkpoint
     if resume is not None:
```
Comment (rahelbeloch) [inline train.py]: This is where the model loading is already happening.

File: train.py [hunk 3]
```diff
@@ -62,6 +63,8 @@ def train(model_name, seed, epochs, patience, b_size, l_rate, w_decay, warmup, m
     else:
         model = DocumentClassifier(model_params, optimizer_hparams)
 
+    test_acc, val_acc = evaluate(trainer, model, test_loader, val_loader)
```
Comment (rahelbeloch) [inline train.py:66]: So you do

- evaluate
- train
- evaluate

right? Is this the way to go with the transfer?

**Review by rahelbeloch — CHANGES_REQUESTED** [review]
@MatPrst I made some comments. Let me know what you think about these.

File: train.py (hunk 3 above)
Comment (matprst) [reply train.py:66]: It's was mostly to see whether what as the initial accuracy without training. Not related to transfer

File: models/model.py (hunk 1 above)
Comment (matprst) [reply models/model.py:34]: Why do we need to resume the training in a normal training setting (without transfer)? There is a difference between resuming and transferring because for transfer we freeze the weights of the encoder

Comment (rahelbeloch) [reply models/model.py:34]: I think it was more like a "we need something to reload pretrained models". But then we need the preloaded models for two cases: 1. transfer and 2. Roberta GNN (which is not implemented yet). So maybe we could have 1 parser argument for the checkpoint and 1 parser argument for the "mode" for which we load it (either transfer or default the latter would simply load the checkpoint and do no freezing).

File: train.py (hunk 3 above)
Comment (rahelbeloch) [reply train.py:66]: Ah ok!

File: models/model.py (hunk 1 above)
Comment (matprst) [reply models/model.py:34]: Good idea, I will make a ```transfer``` argument which will be a flag to set when transferring.

**Review by rahelbeloch — APPROVED** [review]
